File: src/main/java/org/dbshell/commands/connections/SqlValueProvider.java
package org.dbshell.commands.connections;

import com.github.mnadeem.TableNameParser;
import org.springframework.core.MethodParameter;
import org.springframework.shell.CompletionContext;
import org.springframework.shell.CompletionProposal;
import org.springframework.shell.standard.ValueProviderSupport;
import org.springframework.stereotype.Component;

import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

@Component
public class SqlValueProvider extends ValueProviderSupport {
    @Override
    public List<CompletionProposal> complete(MethodParameter parameter, CompletionContext completionContext, String[] hints) {
        String currentInput = completionContext.currentWordUpToCursor();
        return new TableNameParser(currentInput).tables()
                .stream()
                .filter(s -> s.contains(currentInput))
                .map(CompletionProposal::new)
                .collect(Collectors.toList());



    }
}

File: src/main/kotlin/org/dbshell/providers/SqlValueProvider.kt
package org.dbshell.providers

import com.github.mnadeem.TableNameParser
import org.springframework.core.MethodParameter
import org.springframework.shell.CompletionContext
import org.springframework.shell.CompletionProposal
import org.springframework.shell.standard.ValueProvider
import org.springframework.stereotype.Component

@Component
class SqlValueProvider: ValueProvider {
    override fun complete(
        completionContext: CompletionContext
    ): MutableList<CompletionProposal> {

        val currentInput = completionContext.currentWordUpToCursor()
        return TableNameParser(currentInput).tables()
            .filter{s -> s.contains(currentInput!!)}
            .map{cp -> CompletionProposal(cp)}
            .toMutableList()
    }
}